tests/lib/rules/no-identical-names.js

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-template-curly-in-string */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/no-identical-names"),
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+
+ruleTester.run("no-identical-title", rule, {
+
+    valid: [
+        [
+            "module(\"module\");",
+            "test(\"test\", function() {});"
+        ].join("\n"),
+        [
+            "module(\"module1\");",
+            "test(\"it1\", function() {});",
+            "test(\"it2\", function() {});"
+        ].join("\n"),
+        [
+            "test(\"it1\", function() {});",
+            "test(\"it2\", function() {});"
+        ].join("\n"),
+        [
+            "module(\"title\", function() {});",
+            "test(\"title\", function() {});"
+        ].join("\n"),
+        [
+            "module(\"module1\");",
+            "test(\"it1\", function() {});",
+            "module(\"module2\");",
+            "test(\"it1\", function() {});"
+        ].join("\n"),
+        [
+            "module(\"module1\");",
+            "module(\"module2\");"
+        ].join("\n"),
+        [
+            "test(\"test\" + n, function() {});",
+            "test(\"test\" + n, function() {});"
+        ].join("\n"),
+        {
+            code: [
+                "test(`it${n}`, function() {});",
+                "test(`it${n}`, function() {});"
+            ].join("\n"),
+            env: {
+                es6: true
```

Comment:
I think I'd prefer `parserOptions: { ecmaVersion: 6 }` here-- setting the es6 environment also enables some globals and other things and I'd rather just focus on the parsing capability of allowing template strings. Sorry for not catching this the first time around.